liquibase-core/src/main/java/liquibase/lockservice/ProlongingLockService.java

```diff
@@ -0,0 +1,879 @@
+import liquibase.database.core.MSSQLDatabase;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.LockException;
+import liquibase.exception.UnexpectedLiquibaseException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.logging.LogFactory;
+import liquibase.snapshot.InvalidExampleException;
+import liquibase.snapshot.SnapshotGeneratorFactory;
+import liquibase.sql.Sql;
+import liquibase.sqlgenerator.SqlGeneratorFactory;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.AddColumnStatement;
+import liquibase.statement.core.CreateDatabaseChangeLogLockTableStatement;
+import liquibase.statement.core.DropTableStatement;
+import liquibase.statement.core.InitializeDatabaseChangeLogLockTableStatement;
+import liquibase.statement.core.LockDatabaseChangeLogStatement;
+import liquibase.statement.core.ProlongDatabaseChangeLogLockStatement;
+import liquibase.statement.core.RawSqlStatement;
+import liquibase.statement.core.RemoveStaleLocksStatement;
+import liquibase.statement.core.SelectFromDatabaseChangeLogLockStatement;
+import liquibase.statement.core.UnlockDatabaseChangeLogStatement;
+import liquibase.structure.core.Table;
+
+/**
+ * Prolongs the current lock every 30 seconds.
+ * <p>
+ * Locks older than 100 seconds are considered stale and will get removed.
+ * <p>
+ * Since locks get actively prolonged, do use a very high lock wait time (1 day).
+ * <p>
+ * Must <b>not</b> be used in junction with the {@link StandardLockService}!
+ * <p>
+ * Note: This is in alpha state. We do not use the time of the database server right now,
+ * but a local date, so this might not work properly when several services with very
+ * un-synchronised locks connect to the same database.
+ */
+public class ProlongingLockService implements LockService {
+
+    private static final int LOW_PRIO = -1;
+    private static final int PRIO_HIGHER_THAN_STANDARD_LOCK_SERVICE = 2;
+
+    // Only prolong your own lock, hence use a unique id for each service
+    // accessing the database
+    private static final String LOCKED_BY_ID = UUID.randomUUID().toString();
+
+    protected Database database;
+
+    /**
+     * This is a local timestamp. We record when our lock would be stale.
+     * <p>
+     * On the server, however, we use current_timestamp instead.
+     * <p>
+     * So these two values could be different, but as long as we compare
+     * values on the server with each other, and local Dates with each other,
+     * it's not a problem.
+     * <p>
+     * Treat this as the local, cached version of the value that is written
+     * in LOCKEXPIRES in the database.
+     */
+    private volatile Date lockExpires = null;
+
+    private volatile ScheduledFuture<?> lockProlonger = null;
+
+    /**
+     * How long shall we wait for a lock before we give up?
+     * <p>
+     * Unit: minutes
```

Comment:
Better to use a `java.time.Duration`? 
Time fields without units in their names are conventionally seconds (but I prefer putting units in the name, e.g. `changeLogWaitTimeMinutes`)

Reply:
Can't use duration here, it came with java 1.8 but we have 1.6 here.
Changing the name sounds like a good idea.